jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/DockerContextTask.java

```diff
@@ -64,18 +65,18 @@ public void generateDockerContext() {
     Preconditions.checkNotNull(jibExtension);
 
     GradleBuildLogger gradleBuildLogger = new GradleBuildLogger(getLogger());
-
-    ProjectProperties projectProperties =
-        ProjectProperties.getForProject(getProject(), gradleBuildLogger);
-    String mainClass = projectProperties.getMainClass(jibExtension.getMainClass());
+    GradleProjectProperties gradleProjectProperties =
+        GradleProjectProperties.getForProject(getProject(), gradleBuildLogger);
 
     String targetDir = getTargetDir();
 
     try {
-      new DockerContextGenerator(projectProperties.getSourceFilesConfiguration())
+      new DockerContextGenerator(gradleProjectProperties.getSourceFilesConfiguration())
           .setBaseImage(jibExtension.getBaseImage())
           .setJvmFlags(jibExtension.getJvmFlags())
-          .setMainClass(mainClass)
+          .setMainClass(
+              MainClassFinder.resolveMainClass(
```

Comment:
I think we should be explicit about the ordering of log/error messages that show up. We should probably keep all the method calls that could error/log things together near the top before any logic that works on the return values of those methods. This way, we can easily see what order things could be checked in.